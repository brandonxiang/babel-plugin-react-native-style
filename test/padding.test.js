const pluginTester = require('babel-plugin-tester');
const reactNativeStyle = require('../src/index');

pluginTester({
  plugin: reactNativeStyle,
  pluginName: 'babel-plugin-react-native-style',

  tests: [
    {
      snapshot: true,
      title: 'padding test',
      pluginOptions: { rpx: { enable: true, size: 750 } },
      code: `
      const styles = stylesheet.create({
        one: { padding: '0pt' },
        two: { padding: '10% 1pt' },
        three: { padding: '10% 1pt 5' },
        four: { padding: '10% 1pt 0.5 3' },
        five: { padding: '10 auto 5' },
      })
    `,
    },
  ],
});
