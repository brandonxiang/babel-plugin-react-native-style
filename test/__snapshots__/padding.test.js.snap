// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel-plugin-react-native-style padding test: padding test 1`] = `
"
const styles = stylesheet.create({
  one: { padding: '0pt' },
  two: { padding: '10% 1pt' },
  three: { padding: '10% 1pt 5' },
  four: { padding: '10% 1pt 0.5 3' },
  five: { padding: '10 auto 5' },
})

      ↓ ↓ ↓ ↓ ↓ ↓

import { __RPX } from 'babel-plugin-react-native-style/src/runtime/index';
const styles = stylesheet.create({
  one: {
    paddingTop: 0,
    paddingRight: 0,
    paddingBottom: 0,
    paddingLeft: 0
  },
  two: {
    paddingTop: '10%',
    paddingRight: 1,
    paddingBottom: '10%',
    paddingLeft: 1
  },
  three: {
    paddingTop: '10%',
    paddingRight: 1,
    paddingBottom: __RPX(5, 750),
    paddingLeft: 1
  },
  four: {
    paddingTop: '10%',
    paddingRight: 1,
    paddingBottom: __RPX(0.5, 750),
    paddingLeft: __RPX(3, 750)
  },
  five: {
    paddingTop: __RPX(10, 750),
    paddingRight: 'auto',
    paddingBottom: __RPX(5, 750),
    paddingLeft: 'auto'
  }
});
"
`;
